browser/ui/views/tabs/brave_tab_container.cc

```diff
@@ -256,6 +263,38 @@ void BraveTabContainer::UpdateLayoutOrientation() {
   InvalidateLayout();
 }
 
+void BraveTabContainer::PaintBoundingBoxForTiles(gfx::Canvas& canvas) {
+  auto* split_view_data =
+      SplitViewBrowserData::FromBrowser(tab_slot_controller_->GetBrowser());
+  base::ranges::for_each(split_view_data->tiles(), [&](const auto& tile) {
+    PaintBoundingBoxForTile(canvas, tile);
+  });
+}
+
+void BraveTabContainer::PaintBoundingBoxForTile(
+    gfx::Canvas& canvas,
+    const SplitViewBrowserData::Tile& tile) {
+  auto* tab_strip_model = tab_slot_controller_->GetBrowser()->tab_strip_model();
+  auto tab1_index = tab_strip_model->GetIndexOfTab(tile.first);
+  auto tab2_index = tab_strip_model->GetIndexOfTab(tile.second);
+  CHECK(controller_->IsValidModelIndex(tab1_index));
+  CHECK(controller_->IsValidModelIndex(tab2_index));
+
+  gfx::Rect bounding_rects;
+  for (auto i : {tab1_index, tab2_index}) {
+    bounding_rects.Union(GetTabAtModelIndex(i)->bounds());
+  }
+  // TODO(sko) This might need fine tune with layout adjustments.
+  bounding_rects.Inset(gfx::Insets::VH(1, 0));
+
+  constexpr auto kRadius = 12.f;  // same value with --leo-radius-l
+
+  cc::PaintFlags flags;
+  auto* cp = GetColorProvider();
```

Comment:
don't need to check |cp| is valid?

Reply:
Not sure. Can it be null inside OnPaint()?

Reply:
Let's add CHECK() then 😁

Reply:
Well... it seems painting is not guarded `GetWidget()`

Reply:
why a CHECK? We should crash in this case?

Reply:
I remember @fallaciousreasoning and I talking based on the upstream doc https://source.chromium.org/chromium/chromium/src/+/main:styleguide/c++/checks.md;l=75

They recommend CHECK() over DCHECK() when checking invariant unless the check is inexpensive. But from my experience for about 1 yr, I still prefer checking invariant with DCHECK() due to the same reasone you noted: we don't want crash, and just want to check the invariant during the development. Do you think it's okay to ignore their recommendation?